### DISCUSSION TITLE
Importing package dependancies / node_modules

### DISCUSSION THREAD

**Discussion by ytsruh** [nodejs/single-executable#70] [Q&A] [answered]
Hi all,

I've been playing arounf with this and firslty I just wanted to say congratulations - this is a huge step for NodeJs.

However, I'm struggling to get any files to import. I've successfully managed to build an SEA using just a single file. But I'm unables to either:
a) import my own JS files into the main script im using as the build entry point
b) import packages that have been installed using `npm install`

I have read the docs & tried to include the following code, without success:
```
const { createRequire } = require('node:module');
require = createRequire(__filename);
```

Any help is super appreciated

**Comment by aduh95** [comment, marked as answer]
> The single executable application feature currently only supports running a single embedded script using the [CommonJS](https://nodejs.org/api/modules.html#modules-commonjs-modules) module system.

Source: https://nodejs.org/api/single-executable-applications.html

It's expected it won't work, you'd need to bundle all your code and don't rely on external JS modules.

**Reply by ytsruh** [reply]
Understood & thanks a lot for the explanation

**Comment by ghost** [comment]
> > The single executable application feature currently only supports running a single embedded script using the [CommonJS](https://nodejs.org/api/modules.html#modules-commonjs-modules) module system.
> 
> Source: https://nodejs.org/api/single-executable-applications.html
> 
> It's expected it won't work, you'd need to bundle all your code and don't rely on external JS modules.

Is there any way to currently get around this ?
Edit: I've used esbuild.

**Comment by HeyITGuyFixIt** [comment]
> It's expected it won't work, you'd need to bundle all your code and don't rely on external JS modules.

I bundled the majority of my dependencies with esbuild, but I am depending on the canvas module, which doesn't seem to be able to be bundled, because of the canvas.node and DLL/SO files that it downloads. Are there any plans to allow for some means to rely on external JS modules?

**Reply by RaisinTen** [reply]
I think eventually, the native addons would become a part of the VFS.

**Comment by maelp** [comment]
Is there a way to create a standalone executable for some given OS when using a package that requires FFI, like @abandonware/noble ?
